src/namespace.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * namespace is a generic interface for global tram-one state that needs
+ * to be persisted in the app container. It exposes a setup and get function.
+ */
+
+export const setupTramOneSpace = () => {
+	window['tram-space'] = {};
+};
+
+export const buildNamespace = <NamespaceStore>(constructor: () => NamespaceStore) => {
```

Comment:
In order to have the `get` `set` and `setup` functions all produce the same Type, we now have one generic function whose type is inferred by the constructor that is used.